lib/block_view_dao_coin_limit_order.go

```diff
@@ -1062,16 +1066,147 @@ func (bav *UtxoView) _getDAOCoinLimitOrderEntry(inputEntry *DAOCoinLimitOrderEnt
+	for _, orderEntry := range bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry {
+		if orderEntry.BuyingDAOCoinCreatorPKID.Eq(buyingDAOCoinCreatorPKID) &&
+			orderEntry.SellingDAOCoinCreatorPKID.Eq(sellingDAOCoinCreatorPKID) {
+			// Note: we can't just take ToMapKey() here as that does
+			// not include the quantity or block height in the struct.
+			orderEntryBytes, err := orderEntry.ToBytes()
+			if err != nil {
+				// This should never happen as DAO coin limit
+				// order ToBytes() never returns an error.
+				return nil, err
+			}
+			orderEntryString := string(orderEntryBytes)
+
+			outputEntries = append(outputEntries, orderEntry)
+			seenOrderEntries[orderEntryString] = true
+		}
+	}
+
+	// Get orders from the database.
+	dbAdapter := DbAdapter{
+		badgerDb:   bav.Handle,
+		postgresDb: bav.Postgres,
+	}
+
+	dbOrderEntries, err := dbAdapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(
+		buyingDAOCoinCreatorPKID, sellingDAOCoinCreatorPKID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, orderEntry := range dbOrderEntries {
+		// Note: we can't just take ToMapKey() here as that does
+		// not include the quantity or block height in the struct.
+		orderEntryBytes, err := orderEntry.ToBytes()
+		if err != nil {
+			// This should never happen as DAO coin limit
+			// order ToBytes() never returns an error.
+			return nil, err
+		}
+		orderEntryString := string(orderEntryBytes)
+
+		if _, exists := seenOrderEntries[orderEntryString]; !exists {
+			outputEntries = append(outputEntries, orderEntry)
+			seenOrderEntries[orderEntryString] = true
+		}
+	}
+
+	return outputEntries, nil
+}
+
+func (bav *UtxoView) _getAllDAOCoinLimitOrdersForThisTransactor(transactorPKID *PKID) ([]*DAOCoinLimitOrderEntry, error) {
+	// This function is used by the API to construct all open orders for the input transactor.
+	if transactorPKID == nil {
+		return nil, errors.Errorf("_getAllDAOCoinLimitOrdersForThisTransactor: Called with nil transactor PKID; this should never happen")
+	}
+
+	outputEntries := []*DAOCoinLimitOrderEntry{}
+
+	// We want to only return a unique list of orders. We use this map as
+	// a Set to dedup orders between the UTXO view and the database with
+	// preference given to orders from the UTXO view as they are newer.
+	seenOrderEntries := map[string]bool{}
+
+	// Get orders from the UTXO view.
+	for _, orderEntry := range bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry {
+		if orderEntry.TransactorPKID.Eq(transactorPKID) {
+			// Note: we can't just take ToMapKey() here as that does
+			// not include the quantity or block height in the struct.
+			orderEntryBytes, err := orderEntry.ToBytes()
+			if err != nil {
+				// This should never happen as DAO coin limit
+				// order ToBytes() never returns an error.
+				return nil, err
+			}
+			orderEntryString := string(orderEntryBytes)
+
+			outputEntries = append(outputEntries, orderEntry)
+			seenOrderEntries[orderEntryString] = true
+		}
+	}
```

Comment:
Generally the process we follow for these functions is: 
1. Get everything from the DB first
2. Merge into view
3. Iterate over view and get the stuff you need
As an example, see `GetNFTEntriesForPostHash` on line 63 of `block_view_nft.go`